src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Embedded.cs

```diff
@@ -0,0 +1,435 @@
+
+            var code = @"
+[assembly: System.Runtime.CompilerServices.TypeForwardedToAttribute(typeof(Microsoft.CodeAnalysis.EmbeddedAttribute))]
+class Test
+{
+    public void M(ref readonly int p)
+    {
+        // This should trigger generating another EmbeddedAttribute
+    }
+}";
+
+            CreateCompilationWithMscorlib(code, references: new[] { reference }).VerifyEmitDiagnostics(
+                // error CS8006: Forwarded type 'EmbeddedAttribute' conflicts with type declared in primary module of this assembly.
+                Diagnostic(ErrorCode.ERR_ForwardedTypeConflictsWithDeclaration).WithArguments("Microsoft.CodeAnalysis.EmbeddedAttribute").WithLocation(1, 1));
+        }
+
+        [Fact]
+        public void CompilerShouldIgnorePublicEmbeddedAttributesInReferencedAssemblies()
+        {
+            var reference = CreateCompilationWithMscorlib(assemblyName: "testRef", text: @"
+namespace Microsoft.CodeAnalysis
+{
+    public class EmbeddedAttribute : System.Attribute { }
+}
+namespace OtherNamespace
+{
+    public class TestReference { }
+}").ToMetadataReference();
+
+            var code = @"
+class Test
+{
+    // This should trigger generating another EmbeddedAttribute
+    public void M(ref readonly int p)
+    {
+        var obj = new OtherNamespace.TestReference(); // This should be fine
+    }
+}";
+
+            CompileAndVerify(code, verify: false, additionalRefs: new[] { reference }, symbolValidator: module =>
+            {
+                var attributeName = AttributeDescription.CodeAnalysisEmbeddedAttribute.FullName;
+
+                var referenceAttribute = module.GetReferencedAssemblySymbols().Single(assembly => assembly.Name == "testRef").GetTypeByMetadataName(attributeName);
+                Assert.NotNull(referenceAttribute);
+
+                var generatedAttribute = module.ContainingAssembly.GetTypeByMetadataName(attributeName);
+                Assert.NotNull(generatedAttribute);
+
+                Assert.False(referenceAttribute.Equals(generatedAttribute));
+            });
+        }
+
+        [Fact]
+        public void SynthesizingAttributeRequiresSystemAttribute_NonExisting()
+        {
+            var code = @"
+namespace System
+{
+    public class Object {}
+    public class Void {}
+}
+public class Test
+{
+    public void M(ref readonly object x) { } // should trigger synthesizing IsReadOnly
+}";
+
+            CreateCompilation(code).VerifyEmitDiagnostics(CodeAnalysis.Emit.EmitOptions.Default.WithRuntimeMetadataVersion("v4.0.30319"),
+                // error CS0518: Predefined type 'System.Attribute' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Attribute").WithLocation(1, 1),
+                // error CS0518: Predefined type 'System.Attribute' is not defined or imported
+                Diagnostic(ErrorCode.ERR_PredefinedTypeNotFound).WithArguments("System.Attribute").WithLocation(1, 1));
```

Comment:
No. because the compiler is trying to synthesize two attributes here (`Embedded` and `IsReadOnly`).